test/unit/test-helper.sh

```diff
@@ -89,6 +89,16 @@ function mock_gcloud_function() {
   "$@" 2>&1
 }
 
+# Mocks the kubectl functionality (does nothing and outputs nothing).
+# Parameters: $1..$n - parameters passed to the function.
+function mock_kubectl_function() {
+  set -e
+  function gcloud() {
```

Comment:
You're mocking gcloud here, not kubectl.

Reply:
Wow, fixed. Why do the tests still pass with this?

Reply:
Because kubectl does return something in the test, and the function does what's supposed to do. You're just checking for the header, and that's written no matter what pods are in the cluster.